### PR TITLE
coreneuron: undo accidental pinning of @develop

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Review by olupton — CHANGES_REQUESTED** [review]
Obviously fixing the `commit="a1631e6eb1745dc602a522ab8a06e791d13bca13"` part is good.

I think we still need to understand why the CI of this PR is choosing 8.2.2 where the last (relatively?) good pipeline on `develop` chose 9.0.a4: https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/586211.

File: bluebrain/deployment/environments/applications_hpc.yaml [hunk 1]
```diff
@@ -41,13 +41,13 @@ spack:
...
-    - neurodamus-core+common
+    - neurodamus-core~common~~coreneuron
+    - neurodamus-core+common~~coreneuron
     - neurodamus-hippocampus+coreneuron+caliper
     - neurodamus-mousify+coreneuron+caliper
-    - neurodamus-neocortex+coreneuron+caliper
+    - neurodamus-neocortex~plasticity+coreneuron+caliper
     - neurodamus-neocortex+plasticity+coreneuron+caliper
-    - neurodamus-neocortex+ngv+metabolism+caliper
+    - neurodamus-neocortex+ngv+metabolism+caliper~~coreneuron
```
Comment (olupton) [inline bluebrain/deployment/environments/applications_hpc.yaml:50]: I don't have a strong opinion about these. I expect CoreNEURON may be enabled in some of these places soon?

Note to other reviewers: `~~` means it propagates to dependencies.

cc: @pramodk @iomaganaris @WeinaJi

**Review by olupton — APPROVED** [review]

File: bluebrain/repo-bluebrain/packages/coreneuron/package.py [hunk 2]
```diff
@@ -20,9 +20,7 @@ class Coreneuron(CMakePackage):
     # This simplifies testing the gitlab-pipelines repository:
     git = "ssh://git@bbpgitlab.epfl.ch/hpc/coreneuron.git"
 
-    version(
-        "develop", preferred=True, commit="a1631e6eb1745dc602a522ab8a06e791d13bca13"
-    )  # old: branch='master'
+    version("develop", branch="master")
```
Comment (heerener) [inline bluebrain/repo-bluebrain/packages/coreneuron/package.py:23]: It's a nit-pick, but in my upstreaming efforts I've been told that keeping the name of the branch and the version consistent is preferable.

**Review by heerener — APPROVED** [review]

File: bluebrain/repo-bluebrain/packages/coreneuron/package.py (hunk 2 above)
Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/coreneuron/package.py:23]: Yes, we just have "historical reasons" :-/

File: bluebrain/deployment/environments/applications_hpc.yaml [hunk 3]
```diff
@@ -41,13 +41,13 @@ spack:
     # - neuron%nvhpc+tests+gpu+openmp+coreneuron
     # - neuron%nvhpc+tests+gpu+openmp+nmodl+sympy+coreneuron
     - nest@2.20.1
-    - neurodamus-core~common
-    - neurodamus-core+common
+    - neurodamus-core~common~~coreneuron
+    - neurodamus-core+common~~coreneuron
```
Comment (iomaganaris) [inline bluebrain/deployment/environments/applications_hpc.yaml:45]: I think `neurodamus-core` could be removed. We now only depend on it via the submodule in `py-neurodamus`

Comment (matz-e) [reply bluebrain/deployment/environments/applications_hpc.yaml:45]: I see this still:
```
bluebrain/repo-bluebrain/packages/neurodamus-model/package.py
31:    depends_on("neurodamus-core@" + core_v, type="build", when="@" + this_version)
```
Should we then remove the old stuff that uses `version_from_model_core_dep`, too?

Comment (matz-e) [reply bluebrain/deployment/environments/applications_hpc.yaml:45]: If we can clean up the older versions, I think that would be a good start to simplify our package structure.

Comment (iomaganaris) [reply bluebrain/deployment/environments/applications_hpc.yaml:45]: I think everything related to `neurodamus-core` could be removed. We now only depend to it by the submodule of `py-neurodamus`.
I think also we need to remove all the `py-neurodamus` versions prior to `2.9.0` that expect that there is a `neurodamus-core` package and doesn't bring the submodule as well

**Pull request merged**